bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java

```diff
@@ -185,6 +189,7 @@ public static BError createDistinctError(String typeIdName, Module typeIdPkg, BS
     @Deprecated
     public static BError createDistinctError(String typeIdName, Module typeIdPkg, BString message,
                                              BMap<BString, Object> details) {
+        RuntimeUtils.validateErrorDetails(details);
```

Comment:
Some lines seem to be not covered by the test cases. Shall we check these and add tests accordingly?

Reply:
Added test cases to increase the patch coverage.

Reply:
Created a separate issue https://github.com/ballerina-platform/ballerina-lang/issues/35768 to address NPEs occur in master when providing null for error details and value maps.